cider/util/Datum.cpp

```diff
@@ -376,7 +376,11 @@ Datum StringToDatum(const std::string_view s, SQLTypeInfo& ti) {
         d.bigintval = dateTimeParse<kTIMESTAMP>(s, ti.get_dimension());
         break;
       case kDATE:
-        d.bigintval = dateTimeParse<kDATE>(s, ti.get_dimension());
+        if (ti.get_compression() == EncodingType::kENCODING_DATE_IN_DAYS) {
+          d.intval = parseDateInDays(s, ti.get_dimension());
```

Comment:
Why one path is `intval` and the other one is `bigintval`?

Reply:
old codepath use int64 date format(bigintval), while new codepath(flag encoding_in_days) use int32 arrow date format(intval), this will effect runtime func return type.
https://github.com/intel/BDTK/blob/7f18c39f4cddaee713e0c4017081a3f70b0cddb0/cider/exec/template/StringOpsIR.cpp#L163-L172